packages/core/src/network-resolver.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  DataStore,
+  DocumentNetworkResolver,
+  DocumentResolverProps,
+  DocumentResolverResult,
+  WalletDocument,
+} from '@docknetwork/wallet-sdk-data-store/src/types';
+
+type ResolverResult = string | null;
+
+export const dockDocumentNetworkResolver: DocumentNetworkResolver = async ({
+  document,
+  dataStore,
+}: DocumentResolverProps): Promise<DocumentResolverResult> => {
+  let networkId;
+  let isFallback = false;
+
+  for (const resolver of resolvers) {
+    networkId = await resolver({document, dataStore});
+
+    if (networkId) {
+      break;
+    }
+  }
+
+  if (!networkId) {
+    isFallback = true;
+    networkId = dataStore.networkId;
+  }
+
+  return {
+    networkId: networkId,
+    isFallback,
+  };
+};
+
+async function credentialResolver({
+  document,
+}: DocumentResolverProps): Promise<ResolverResult> {
+  if (!document.type.includes('VerifiableCredential')) {
+    return null;
+  }
+
+  if (document.id.indexOf('creds-testnet.dock.io') > -1) {
+    return 'testnet';
+  }
+
+  if (document.id.indexOf('creds.dock.io') > -1) {
+    return 'mainnet';
+  }
+
```

Comment:
do we need to fall back to a DID resolution check here?

Reply:
I think so; I've added a TODO for it and will implement it along with the account resolver

Reply:
should we allow a mapping between hostname and network to be injected with the network configs? Just thinking of how we would do this for other networks. Maybe have the network definition be something like

```
    {
      name: 'polygon',
      id: 'polygon',
      credential-urls: [],
      configs: '{}',
    }
```

and then this resolver would just look for a match in the array of linked credential-urls

Reply:
that's a good idea, looks very flexible. Let me add this to this PR

Reply:
Ideally we won't have 'mainnet' or 'dock.io' or anything like that hard-coded in either app when we're done